src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,258 @@
+import { useEffect, useState, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { ErrorOption, Person } from '../types';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { SearchLink } from './SearchLink';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorOption, setErrorOption] = useState(ErrorOption.noError);
+  const PEOPLE_URL
+  = 'https://mate-academy.github.io/react_people-table/api/people.json';
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const resetAll = () => {
+    return {
+      centuries: null,
+      query: null,
+      sex: null,
+      sort: null,
+      order: null,
+    };
+  };
+
+  const getCentury = (bornYear: number) => Math.ceil(bornYear / 100);
+
+  const generateClass = (sortBy: string) => {
+    return cn('fas fa-sort', {
+      'fa-sort-up': sort === sortBy && !order,
+      'fa-sort-down': sort === sortBy && order,
+    });
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (event.target.value) {
+      params.set('query', event.target.value);
+    } else {
+      params.delete('query');
+    }
```

Comment:
make sense to use ternary operator here as it will make a code shorter

Reply:
https://stackoverflow.com/questions/18060002/ternary-operator-displays-error-in-jshint-expected-an-assignment-or-function-c